api/caasunitprovisioner/client.go

```diff
@@ -81,15 +81,15 @@ func (c *Client) ApplicationConfig(applicationName string) (application.ConfigAt
 // WatchUnits returns a StringsWatcher that notifies of
 // changes to the lifecycles of units of the specified
 // CAAS application in the current model.
-func (c *Client) WatchUnits(application string) (watcher.StringsWatcher, error) {
+func (c *Client) WatchApplicationScale(application string) (watcher.NotifyWatcher, error) {
```

Comment:
Worth changing folder name from 'caasunitprovisioner'? Maybe a 'caasapplicationscaler'?

Reply:
It also does storage etc, so provisioner is generic here